### ISSUE TITLE
[tw_gcal_sync] [Bug] Timezone difference

### ISSUE THREAD

**Issue by Petemir** [bergercookie/syncall#35]
Hello,

I have been having a problem when syncing tasks to a calendar. e.g., on my `pending.data` file the event has as a due timestamp `1649250000`, which on my current timezone (GMT+2 -DST-, changed from GMT+1 on 27/03/2022) translates to today at 15:00:00 (according to [this site](https://www.epochconverter.com/)). My Google Calendar is also set with GMT+2 as the timezone. Yet, when I sync the event, I actually see it at 14:00:00 on the Calendar. 

This also happened with the events I created while on GMT+1 (which, funnily enough, they *now* show at the correct time on the calendar).

Let me know if there's anything else I can help with to debug this.

Cheers!

**Comment by Chantico360** [comment]
Hello,

I'm also affected by this bug. I tried every combination from system date/timezone and calender date/timezone settings and didn't found any working combination. I modified some events multiple times and they've got magically the correct time, but I don't how it came. On creation it never worked for me.

It really impacts my usage of the (otherwise great!) tool, because if I can't rely on the date it's kind of useless.. :/

**Comment by bergercookie** [comment]
Hi @Chantico360 , @Petemir , I'll try to address this during the next week.
Stay tuned. :)

**Comment by Chantico360** [comment]
Thank you very much! That's great to hear! :)

**Comment by Petemir** [comment]
Cool! Just in time for the DST ending in Europe ;). Thank you!

**Comment by bergercookie** [comment]
> Hi @Chantico360 , @Petemir , I'll try to address this during the next week.

Lol, gross miscalculation 😅 

In any case, this is defo an issue and I'll be looking to address this. In the meantime, do feel free to send a PR if it's pressing for you.

**Comment by espg** [comment]
...I'm curious if there's a different cause for this bug. I've been running into the same thing, and I think that it's actually because of differences in how TW versus google calendar handles the interval. 

If I set something in task warrior with `due:2023-02-07T19:00` , it shows up an hour offset (one hour later in google); same if I put something in google at 7pm, one hour offset after sync to taskwarrior.

I noticed something when setting a few non-standard meeting times (not the default hour block) in google... if I set a half hour meeting, then it's only off by a half hour in taskwarrior. Setting a long duration event in google (like a flight) yields a longer offset.

It seems like when you sync with google to taskwarrior, it will take the *end* of the duration as the 'due' date/time. So if you mark an hour long meeting, it will set it as 'due' when the meeting ends, instead of when it starts. The same thing happens in reverse when syncing the other direction-- the start time is calculated off the due time using the assumed default duration.

@bergercookie  is there a simple way to override this behavior using existing config options/files, or does need a PR to change?
